spec/02-integration/03-db/02-db_core_entities_spec.lua

```diff
@@ -1951,5 +1954,27 @@ for _, strategy in helpers.each_strategy() do
         end)
       end)
     end)
+
+
+    describe("SNIs", function()
+      local certificate
+      lazy_setup(function()
+        certificate = db.certificates.schema:extract_pk_values(bp.certificates:insert())
+      end)
+      describe(":list_for_certificate()", function()
+        it("returns array with cjson.array_mt metatable", function()
+          local snis = assert(db.snis:list_for_certificate(certificate))
+          local json = cjson.encode(snis)
+          assert.equal("[]", json)
+          assert.equal(cjson.array_mt, getmetatable(snis))
+
+          db.snis:update_list(certificate, {"example.org"})
+          snis = assert(db.snis:list_for_certificate(certificate))
+          json = cjson.encode(snis)
+          assert.equal('["example.org"]', json)
+          assert.equal(cjson.array_mt, getmetatable(snis))
+        end)
```

Comment:
This test passes with the old `cjson.empty_array_mt` metatable, so long as we remove the `assert.equal(cjson.array_mt, ...)` assertions. I wonder if this is a fix for a known issue (in which we don't JSON-encode the result properly) with an incomplete regression test, or a refactor to use `array_mt` (in which case, it should be bundled with #4724)?

Reply:
@thibaultcha, the other functions reset the metatable. I will bundle it to #4724, and rebase that to master. I will close this.